### PR TITLE
chore(deps): update all minor and patch deps, remove lint errors

### PR DESCRIPTION
**Pull request by TheLarkInn**
**What kind of change does this PR introduce?**
Refactor, feature.

**Did you add tests for your changes?**
N/A

**If relevant, link to documentation update:**

**Summary**
This PR: 
* updates minimum NodeJS Requirements to 6.11.5
* updates node-eslint-plugin to check against package.json "engines" property
* update engines prop in package.json
* update all dependencies to install latest minor and patch level changes

Since we are dropping node 4 support in webpack v4, this is being applied against the next branch.
**Does this PR introduce a breaking change?**
Yes.

This is a hard requirement for webpack/webpack#next branch (and eventually master) to drop node 4 as LTS has ended for it. 

**Other information**

### CODE REVIEW

File: test/mocha.opts [hunk 1]
```diff
@@ -1,2 +1,3 @@
 --full-trace
 --reporter dot
+--globals Base64
```
Comment (sokra) [inline test/mocha.opts]: That's weird, why is this needed?

Comment (TheLarkInn) [reply test/mocha.opts]: I couldn't figure out where the heck it is coming from. I can try and dig in more. Would it come from a dependent package?! That is the only thing I can think of.

Comment (sokra) [reply test/mocha.opts]: Yep, looks like a dependency does `Base64 = ...` instead of `var Base64 = ...`. Could you check where this is coming from? (search for `Base64` in `node_modules`) And send a PR to the broken package. I can't accept this change.

Comment (sokra) [reply test/mocha.opts]: Caused by https://github.com/dankogai/js-base64/commit/39b64ba0316f71f438cacab4b821b1947cd0f491

cc @dankogai

Comment (sokra) [reply test/mocha.opts]: `postcss@5` depends on `js-base64`.

**Comment by TheLarkInn** [comment]
I'll try and recommit these with different account. Having issues ssince I setup my new computer for this XD

**Comment by sokra** [comment]
Thanks

File: package.json [hunk 2]
```diff
@@ -69,7 +69,7 @@
     "worker-loader": "^0.8.0"
   },
   "engines": {
-    "node": ">=4.3.0 <5.0.0 || >=5.10"
+    "node": ">=6.11.5"
```
Comment (ematipico) [inline package.json]: Is webpack 4 going to remove drop support of node v4?

Comment (TheLarkInn) [reply package.json]: Yes

Comment (brutalcrozt) [reply package.json]: Why? Seems too soon.

Release | LTS Status | Codename | Active LTS Start | Maintenance LTS Start | Maintenance LTS End
-- | -- | -- | -- | -- | --
4.x | Maintenance LTS | Argon | 2015-10-01 | 2017-04-01 | April 2018
